src/client/reducers/audioReducer.js

```diff
@@ -0,0 +1,67 @@
+import { CREATE_AUDIO_DATA, PLAY_PAUSE_SOUND_FROM_FILE, CREATE_STREAME_DATA, START_MUTE_STREAME_AUDIO, MERGE_CANVAS_WIDTH } from '../actions/types';
+
+const context = new (window.AudioContext || window.webkitAudioContext)();    
+const analyser = context.createAnalyser();
+
+const initialState = {  
+    //graphic canvas 
+     widthCanvas: 500,
+     heightCanvas: 150, 
+    //audio from file
+     trackName: undefined,     
+     trackType: undefined,
+     trackSize: undefined,     
+     audioContext: context,
```

Comment:
You need to create an action that will change audio context, and changing it directly is forbidden

Reply:
I made action and avoid it in componentDidMount method (Equalizer component) but when component render it will no see this prop in store (it says that it is undefined).

Reply:
it was undefined because you used it before you actually add it to store, so that is why its undefined 
because didMount will be called after all functions will be called and component is actually mounted to DOM
so just call your action before you use property that doesn`t exist yet

Reply:
this properties (audiocontext and analyzer) will be used in many components (sliders with effects, volume, etc..). They must be in store as base properties after when user come on route .../main. 
Before componentDidMount there is constructor (I can not put there action). Maybe I must put this action in another component, that it render before analyzer?